.github/workflows/python-package.yml

```diff
@@ -17,150 +17,373 @@ on:
+      run: cmake --build ${{ steps.strings.outputs.build-output-dir }} --config ${{ matrix.build_type }}
+
+    - name: Copy libraries
+      shell: bash
+      run: |
         mkdir -p output/${{ matrix.os }}/${{ matrix.arch }}
         ( shopt -s nullglob && cp bitsandbytes/*.{so,dylib,dll} output/${{ matrix.os }}/${{ matrix.arch }}/ )
     - name: Upload build artifact
       uses: actions/upload-artifact@v4
       with:
         name: shared_library_${{ matrix.os }}_${{ matrix.arch }}
         path: output/*
-        retention-days: 7
+
   ##
-  # This job matrix builds the CUDA versions of the libraries for platforms that support CUDA (Linux x64/aarch64 + Windows x64)
+  # This job matrix builds the CUDA versions of the libraries for platforms that support CUDA (Windows)
   ##
   build-shared-libs-cuda:
     strategy:
+      # Set fail-fast to false to ensure that feedback is delivered for all matrix combinations. Consider changing this to true when your workflow is stable.
+      fail-fast: false
+
       matrix:
-        os: [ubuntu-latest, windows-latest]
-        arch: [x86_64, aarch64]
-        cuda_version: ['12.1.0']
-        exclude:
-          - os: windows-latest # This probably requires arm64 Windows agents
-            arch: aarch64
+        os: [windows-latest]
+        arch: [x86_64]
+        cuda-version: ['11.8.0', '12.1.1']
+        build_type: [Release]
     runs-on: ${{ matrix.os }} # One day, we could run them on native agents. Azure supports this now but it's planned only for Q3 2023 for hosted agents
     steps:
       # Check out code
     - uses: actions/checkout@v4
       # Linux: We use Docker to build cross platform Cuda (aarch64 is built in emulation)
-    - name: Set up Docker multiarch
-      if: startsWith(matrix.os, 'ubuntu')
-      uses: docker/setup-qemu-action@v2
-      # On Linux we use CMake within Docker
     - name: Setup cmake
-      if: ${{ !startsWith(matrix.os, 'linux') }}
       uses: jwlawson/actions-setup-cmake@v1.14
       with:
         cmake-version: '3.26.x'
-      # Windows: We install Cuda on the agent (slow)
-    - uses: Jimver/cuda-toolkit@v0.2.14
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.10"
+
+    - name: Set up MSVC
       if: startsWith(matrix.os, 'windows')
-      id: cuda-toolkit
-      with:
-        cuda: ${{ matrix.cuda_version }}
-        method: 'local'
-        # sub-packages: '["nvcc","cudart","nvrtc_dev","cublas_dev","cusparse_dev","visual_studio_integration"]'
-    - name: Add msbuild to PATH
-      uses: microsoft/setup-msbuild@v1.1
-      if: ${{ startsWith(matrix.os, 'windows') }}
+      uses: ilammy/msvc-dev-cmd@v1.13.0
       # Check out dependencies code
     - uses: actions/checkout@v4
       name: Check out NVidia cub
       with:
         repository: nvidia/cub
         ref: 1.11.0
         path: dependencies/cub
-      # Compile C++ code
-    - name: Build C++
+    - name: Setup Mambaforge
```

Comment:
I am still not convinced mamba is the way to go.

Also, there's a lot of duplication here w.r.t the Docker flow. What is the reason we need to change this?

Reply:
Same, it seems overly complicated for not much obvious benefit to me. I like to be able to do something simple like `pip install -e .[dev]` and use the cache on it.